applications/trilinos_application/trilinos_space.h

```diff
@@ -564,7 +564,7 @@ class TrilinosSpace
         rGraph.RowMap().MyGlobalElements(MyGlobalElements);
         
         
-        for(IndexType i=0; i< NumMyRows; i++)
+        for(IndexType i = 0; i < static_cast<IndexType>(NumMyRows); ++i)
```

Comment:
I think the cast is unnecessary here. I would just declare NumMyRows as an IndexType above.
```
IndexType NumMyRows = ....
```
NumMyRows is used to allocate an array in addition to here, so if it is a negative number we have a problem there too.